Features/#54

bugdebugging opened the pull request.

bugdebugging committed:
ENVIRONMENT:Redis Configuration

bugdebugging committed:
FEAT:Redis 도메인 구현

bugdebugging committed:
FEAT:동반러닝 영속화 도메인 Repository

bugdebugging committed:
FIX:Status 까지 지정해서 생성하도록 Result변경

bugdebugging committed:
FIX:Team createdAt필드 삭제

Team의 Goal필드에 started_at이 존재해서 삭제

bugdebugging committed:
FEAT: Goal,Result Equals,HashCode 추가

bugdebugging committed:
TEST:Team,TeamMember Repository Test

pkalsh commented on runmate-domain/src/main/java/com/runmate/domain/running/Goal.java:
```diff
@@ -0,0 +1,30 @@
+
+    @Column(name = "goal_started_at")
+    @CreatedDate
+    private LocalDateTime started_at;
```
카멜 케이스로 변경해주세요

pkalsh commented on runmate-domain/src/main/java/com/runmate/domain/redis/TeamInfo.java:
```diff
@@ -0,0 +1,59 @@
+
+    public boolean isGoalSuccess(Goal goal) {
+        LocalDateTime current = now();
+        LocalDateTime endTime = goal.getStarted_at().plus(goal.getTotalRunningSeconds(), ChronoUnit.SECONDS);
```
Goal 도메인에 endTime 계산하는 로직을 포함시키도록 할 수 있겠네요

pkalsh commented on runmate-domain/src/main/java/com/runmate/domain/running/Team.java:
```diff
@@ -4,13 +4,16 @@
 @Entity
 @Getter
 @NoArgsConstructor(access = AccessLevel.PROTECTED)
+@Table(name = "team")
```
기존 설계의 Team이랑 TeamActivity를 역정규화 하신건가요??

bugdebugging replied:
동반러닝을 수행한다는 것이 물론 Activity를 본인이 참여안해서 없는 경우도 존재하겠지만
거의 data가 있는 경우가 굉장히 많을 거라고 생각해서 정규화를 안해도될거라고 생각했던것같아요.
redis domain 구현에 집중하느라 좀 주의 깊게 구현은 안한거같네요 이부분 좀 더 생각좀 해보겠습니다.

pkalsh replied:
제 생각은 세션때처럼 Room이라는 도메인을 두고 실제로 수행한 팀과 수행하지 않은 팀을 구분하면 역정규화해도 문제 없을 것 같아요

pkalsh reviewed (commented):
리뷰 확인해주세요 :smile:

bugdebugging committed:
FIX:createdAt 필드명 변경

bugdebugging committed:
FEAT:Redis에 저장할 Goal객체 구현

Redis에서 저장하지 않는 다면 Db에서 긇어오거나
Client에서 주는 방법 빼고는 없는데

Db에서 긇어온다면 이는 다른 data를 redis에 올리는 이유를
무색하게 만드는것 db에서 긇어오는 경우가 존재하지 않아야할것으로판단됨.
그래서 판단을 위한 모든 data를 redis에 저장해줘야할것같다.

bugdebugging committed:
FIX:RunningMessage에 초당 달린거리 필드(distance) 추가

bugdebugging committed:
FIX:RunningMessage memberId 필드 추가

bugdebugging committed:
FIX:TeamInfo increaseTotalDistance 매개변수형 변경

bugdebugging committed:
FIX:RunningMessage의 모듈 websocket으로 변경

bugdebugging committed:
FIX:TimeOver 판단 method 추가

bugdebugging committed:
FEAT:TeamMember에 IndividualResult추가

bugdebugging committed:
FIX:목표 체크 메서드 추가

bugdebugging committed:
충돌로 인한 파일 변경

bugdebugging committed:
FIX:Team의 Result를 나중에 결정짓도록 생성자에서 뺌

bugdebugging committed:
FIX:TeamMember의 Result 나중에 결정하도록 변경

bugdebugging committed:
FIX:Goal의 startedAt바로 입력받도록 변경

bugdebugging committed:
TEST:MemberInfo,TeamInfo logic Test
